lang/java/avro/src/main/java/org/apache/avro/io/BinaryEncoder.java

```diff
@@ -37,6 +37,9 @@
  */
 public abstract class BinaryEncoder extends Encoder {
 
+  // Buffer used for writing ASCII strings
```

Comment:
Is the encoder always used in single-threaded way ? 
Until now it is stateless, so it seems it could be used in multi-threaded way too.

Reply:
Hello. Thank you for the review.

I actually borrow this technique from `DirectBinaryEncoder`. It too makes no guarantees around thread safety: none of its methods are synchronized. So I don't think the expectation should be that it is thread safe.

https://github.com/apache/avro/blob/84bc7322ca1c04ab4a8e4e708acf1e271541aac4/lang/java/avro/src/main/java/org/apache/avro/io/DirectBinaryEncoder.java#L44-L45

To your point, maybe we should document this somewhere.

Reply:
A quick look at all the various implementations of `BinaryEncoder`s tells me that none of the Avro ones are thread safe.